4.2-dev4 C# export to Android generates the APK fine, but fails finding .NET assemblies for arm64 at runtime

### Godot version

4.2 dev 4

### System information

mac 12.6.7

### Issue description

![Screenshot_2023-09-20-15-48-42-963_org godotengine godot](https://github.com/godotengine/godot/assets/17978212/186c802d-e33e-49a6-a002-4683deefb3a4)
Why is this? What can I do to fix it?

### Steps to reproduce

.

### Minimal reproduction project

[godot.zip](https://github.com/godotengine/godot/files/12645613/godot.zip)

The same thing here, i cant build the project:

C:\Users\AORUS\Documents\MyRPGGodot\MyRPG-Godot.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.linux-bionic-arm64 with version (= 6.0.18)
       C:\Users\AORUS\Documents\MyRPGGodot\MyRPG-Godot.csproj : error NU1102:   - Found 24 version(s) in nuget.org [ Nearest version: 7.0.0-preview.5.22301.12 ]
       C:\Users\AORUS\Documents\MyRPGGodot\MyRPG-Godot.csproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages

And i cant install the nuget either cause it's not compatible with the project.
OS: WINDOWS 11

Justing to give more information:
I'm using godot 4.2.dev4-mono
.net7.0 as recommended

Error:
       "C:\Users\AORUS\Documents\MyRPGGodot\MyRPG-Godot.csproj" (Restore target) (1) ->
       (Restore target) -> 
         C:\Users\AORUS\Documents\MyRPGGodot\MyRPG-Godot.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.linux-bionic-arm64 with version (= 6.0.18)
       C:\Users\AORUS\Documents\MyRPGGodot\MyRPG-Godot.csproj : error NU1102:   - Found 24 version(s) in nuget.org [ Nearest version: 7.0.0-preview.5.22301.12 ]
       C:\Users\AORUS\Documents\MyRPGGodot\MyRPG-Godot.csproj : error NU1102:   - Found 0 version(s) in Microsoft Visual Studio Offline Packages

    0 Warning(s)
    1 Error(s)

I can't build any architectures of android apk

--- a/modules/mono/csharp_script.cpp
+++ b/modules/mono/csharp_script.cpp
@@ -121,16 +121,16 @@ void CSharpLanguage::init() {
 	GLOBAL_DEF(PropertyInfo(Variant::INT, "dotnet/project/assembly_reload_attempts", PROPERTY_HINT_RANGE, "1,16,1,or_greater"), 3);
 #endif
 
-	gdmono = memnew(GDMono);
-	gdmono->initialize();
-
 #ifdef TOOLS_ENABLED
-	if (gdmono->is_runtime_initialized()) {
-		gdmono->initialize_load_assemblies();
-	}
-
 	EditorNode::add_init_callback(&_editor_init_callback);
 #endif
+
+	gdmono = memnew(GDMono);
+
+	// Initialize only if the project uses C#.
+	if (gdmono->should_initialize()) {
+		gdmono->initialize();
+	}
 }
 
 void CSharpLanguage::finish() {

--- a/modules/mono/editor/GodotTools/GodotTools/Export/ExportPlugin.cs
+++ b/modules/mono/editor/GodotTools/GodotTools/Export/ExportPlugin.cs
@@ -20,6 +20,19 @@ namespace GodotTools.Export
 
         private List<string> _tempFolders = new List<string>();
 
+        private static bool ProjectContainsDotNet()
+        {
+            return File.Exists(GodotSharpDirs.ProjectSlnPath);
+        }
+
+        public override string[] _GetExportFeatures(EditorExportPlatform platform, bool debug)
+        {
+            if (!ProjectContainsDotNet())
+                return Array.Empty<string>();
+
+            return new string[] { "dotnet" };
+        }
+
         public override Godot.Collections.Array<Godot.Collections.Dictionary> _GetExportOptions(EditorExportPlatform platform)
         {
             return new Godot.Collections.Array<Godot.Collections.Dictionary>()
@@ -119,7 +132,7 @@ namespace GodotTools.Export
         {
             _ = flags; // Unused.
 
-            if (!File.Exists(GodotSharpDirs.ProjectSlnPath))
+            if (!ProjectContainsDotNet())
                 return;
 
             if (!DeterminePlatformFromFeatures(features, out string platform))

--- a/modules/mono/mono_gd/gd_mono.cpp
+++ b/modules/mono/mono_gd/gd_mono.cpp
@@ -348,6 +348,15 @@ godot_plugins_initialize_fn try_load_native_aot_library(void *&r_aot_dll_handle)
 
 } // namespace
 
+bool GDMono::should_initialize() {
+#ifdef TOOLS_ENABLED
+	// The editor always needs to initialize the .NET module for now.
+	return true;
+#else
+	return OS::get_singleton()->has_feature("dotnet");
+#endif
+}
+
 static bool _on_core_api_assembly_loaded() {
 	if (!GDMonoCache::godot_api_cache_updated) {
 		return false;
@@ -435,11 +444,15 @@ void GDMono::initialize() {
 
 	_on_core_api_assembly_loaded();
 
+#ifdef TOOLS_ENABLED
+	_try_load_project_assembly();
+#endif
+
 	initialized = true;
 }
 
 #ifdef TOOLS_ENABLED
-void GDMono::initialize_load_assemblies() {
+void GDMono::_try_load_project_assembly() {
 	if (Engine::get_singleton()->is_project_manager_hint()) {
 		return;
 	}

--- a/modules/mono/mono_gd/gd_mono.h
+++ b/modules/mono/mono_gd/gd_mono.h
@@ -72,6 +72,7 @@ class GDMono {
 
 #ifdef TOOLS_ENABLED
 	bool _load_project_assembly();
+	void _try_load_project_assembly();
 #endif
 
 	uint64_t api_core_hash = 0;
@@ -149,10 +150,9 @@ public:
 	Error reload_project_assemblies();
 #endif
 
+	bool should_initialize();
+
 	void initialize();
-#ifdef TOOLS_ENABLED
-	void initialize_load_assemblies();
-#endif
 
 	GDMono();
 	~GDMono();

